ui/camera.js

```diff
@@ -33,32 +33,34 @@ async function initializeModule() {
     }
 }
 
-const ModulePromise = initializeModule().then(initModule => {
-    if (initModule) {
-        return initModule();
-    } else {
-        return null;
+export function rethrowIfCritical(err) {
+    // If it's precisely Error, it's a custom error; anything else - SyntaxError,
+    // WebAssembly.RuntimeError, TypeError, etc. - is treated as critical here.
+    if (err.constructor !== Error) {
+        throw err;
     }
-});
+}
 
 class Camera {
     constructor() {
         /** @type {Promise<unknown>} */
         this.queue = Promise.resolve();
         this.Module = null;
         this.context = null;
-    }
-
-    rethrowIfCritical(err) {
-        // If it's precisely Error, it's a custom error; anything else - SyntaxError,
-        // WebAssembly.RuntimeError, TypeError, etc. - is treated as critical here.
-        if (err.constructor !== Error) {
-            throw err;
-        }
+        this.ModulePromise = null;
```

Comment:
Module is shared between all instances, so this can be a global let variable instead.

Reply:
I'm not exactly sure what you mean with this (and if I already fixed it in later commits). Do you mean there is a single Module object for multiple Camera classes and therefore should be separated from the Camera class? I'm resolving for now, if it is a dealbreaker I'm happy to fix but otherwise I'd say leave it be :)